packages/vite/src/node/ssr/ssrModuleLoader.ts

```diff
@@ -212,26 +212,38 @@ async function instantiateModule(
     )
   } catch (e) {
     mod.ssrError = e
-    const errorData = importErrors.get(e)
 
     if (e.stack && fixStacktrace) {
       ssrFixStacktrace(e, moduleGraph)
     }
 
-    server.config.logger.error(
-      colors.red(
-        `Error when evaluating SSR module ${url}:` +
-          (errorData?.importee
-            ? ` failed to import "${errorData.importee}"`
-            : '') +
-          `\n|- ${e.stack}\n`,
-      ),
-      {
-        timestamp: true,
-        clear: server.config.clearScreen,
-        error: e,
-      },
-    )
+    const originError = errorStacks.get(urlStack)
+
+    const errorDepsStack = originError
+      ? urlStack.concat(originError.importee)
+      : urlStack
+
+    // only log the root url's error
+    if (
+      errorDepsStack.filter((dep) => !dep.startsWith('virtual:'))[0] === url
```

Comment:
Is there a reason this is filtered by `virtual:`? I'm not sure if it's right to handle `virtual:` specifically.

Also, to confirm, is logging the root check helps deduplicate errors?

Reply:
1. Ah I may make a mistake here since `virtual` module should be in the error stack
2. It will prevent the same error from outputting multiple times in a root-leaf path.

Reply:
I reviewed the `virtual:` filtering logic again, the error would not propagate to the virtual module(not sure why), so we have to filter it. @bluwy 

e.g `urlStack` is 
```
['virtual:vite-plugin-ssr:importUserCode:server',
  '/renderer/_error.page.tsx',
  '\x00react/jsx-dev-runtime']
```
The error-catched url would never be` virtual:vite-plugin-ssr:importUserCode:server'`